src/System.ComponentModel.TypeConverter/tests/Design/DesignerOptionServiceTests.cs

```diff
@@ -0,0 +1,487 @@
+        }
+
+        [Fact]
+        public void IndexOf_NoSuchOptions_ReturnsNegativeOne()
+        {
+            var service = new TestDesignerOptionService();
+            DesignerOptionService.DesignerOptionCollection options = service.DoCreateOptionCollection(service.Options, "Name", "Value");
+
+            Assert.Equal(-1, service.Options.IndexOf(service.Options));
+            Assert.Equal(-1, service.Options.IndexOf(null));
+            Assert.Equal(-1, ((IList)service.Options).IndexOf(service.Options));
+        }
+
+        [Fact]
+        public void Contains_ValueExists_ReturnsExpected()
+        {
+            var service = new TestDesignerOptionService();
+            DesignerOptionService.DesignerOptionCollection options = service.DoCreateOptionCollection(service.Options, "Name", "Value");
+
+            Assert.Equal(0, service.Options.IndexOf(options));
+            Assert.Equal(0, ((IList)service.Options).IndexOf(options));
+        }
+
+        [Fact]
+        public void Contains_NoSuchOptions_ReturnsFalse()
+        {
+            var service = new TestDesignerOptionService();
+            IList options = service.DoCreateOptionCollection(service.Options, "Name", "Value");
+
+            Assert.False(options.Contains(service.Options));
+            Assert.False(options.Contains(null));
+        }
+
+        [Theory]
+        [InlineData("Name")]
+        [InlineData("name")]
+        public void Indexer_ValidName_ReturnsExpected(string name)
+        {
+            var service = new TestDesignerOptionService();
+            DesignerOptionService.DesignerOptionCollection options = service.DoCreateOptionCollection(service.Options, "Name", "Value");
+
+            Assert.Same(options, service.Options[name]);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("NoSuchName")]
+        public void Indexer_InvalidName_ReturnsNull(string name)
+        {
+            var service = new TestDesignerOptionService();
+            service.DoCreateOptionCollection(service.Options, "Name", "Value");
+
+            Assert.Null(service.Options[name]);
+        }
+
+        [Fact]
+        public void Indexer_ValidIndex_ReturnsExpected()
+        {
+            var service = new TestDesignerOptionService();
+            DesignerOptionService.DesignerOptionCollection options = service.DoCreateOptionCollection(service.Options, "Name", "Value");
+
+            Assert.Same(options, service.Options[0]);
+            Assert.Same(options, ((IList)service.Options)[0]);
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(1)]
+        public void Indexer_InvalidIndex_ThrowsIndexOutOfRangeException(int index)
+        {
+            var service = new TestDesignerOptionService();
+            service.DoCreateOptionCollection(service.Options, "Name", "Value");
+
+            Assert.Throws<IndexOutOfRangeException>(() => service.Options[index]);
```

Comment:
Duh, I didn't even see the kind of exception you where expecting. Thanks for clarifying.